clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -120,6 +123,8 @@ public Sender(KafkaClient client,
         this.clientId = clientId;
         this.sensors = new SenderMetrics(metrics);
         this.requestTimeout = requestTimeout;
+        
+        this.metadataStaleMs = getMetadataStaleMs(metadata.maxAge(), requestTimeout, metadata.refreshBackoff());
```

Comment:
Just a note. This may need consideration together with #1707 where the metadata age may subject to change during producer startup.